java/ruleXtract/src/main/java/uk/ac/cam/eng/rule/retrieval/RuleRetriever.java

```diff
@@ -378,90 +312,86 @@ public boolean accept(File dir, String name) {
-			retriever.loadDir(params.hfile, conf);
-			retriever.setup(params.test_file, conf);
-			StopWatch stopWatch = new StopWatch();
-			stopWatch.start();
-			System.err.println("Generating query");
-			List<Set<Text>> queries = retriever.generateQueries(
-					params.test_file, conf);
-			System.err.printf("Query took %d seconds to generate\n",
-					stopWatch.getTime() / 1000);
-			System.err.println("Executing queries");
-			try (BufferedWriter out = new BufferedWriter(
-					new OutputStreamWriter(new GZIPOutputStream(
-							new FileOutputStream(params.rules))))) {
-				FeatureCreator features = new FeatureCreator(conf);
-				ExecutorService threadPool = Executors.newFixedThreadPool(Integer.parseInt(params.retrieval_threads));
-
-				for (int i = 0; i < queries.size(); ++i) {
-					HFileRuleQuery query = new HFileRuleQuery(
-							retriever.readers[i], retriever.bfs[i], out,
-							queries.get(i), features, retriever, conf);
-					threadPool.execute(query);
+			Util.parseCommandLine(args, params);
+		} catch (ParameterException e) {
+			return;
+		}
+		RuleRetriever retriever = new RuleRetriever();
+		retriever.loadDir(params.hfile);
+		retriever.setup(params.testFile, params);
+		StopWatch stopWatch = new StopWatch();
+		stopWatch.start();
+		System.err.println("Generating query");
+		List<Set<RuleString>> queries = retriever.generateQueries(
+				params.testFile, params);
+		System.err.printf("Query took %d seconds to generate\n",
+				stopWatch.getTime() / 1000);
+		System.err.println("Executing queries");
+		try (BufferedWriter out = new BufferedWriter(new OutputStreamWriter(
+				new GZIPOutputStream(new FileOutputStream(params.rules))))) {
+			ExecutorService threadPool = Executors
+					.newFixedThreadPool(params.retrievalThreads);
+			for (int i = 0; i < queries.size(); ++i) {
+				HFileRuleQuery query = new HFileRuleQuery(retriever.readers[i],
+						retriever.bfs[i], out, queries.get(i), retriever,
+						params.sp);
+				threadPool.execute(query);
+			}
+			threadPool.shutdown();
+			threadPool.awaitTermination(1, TimeUnit.DAYS);
+			// Add pass through rule not already found in query
+			for (Rule passThroughRule : retriever.passThroughRules) {
+				if (!retriever.foundPassThroughRules.contains(passThroughRule)) {
+					retriever.writeRule(EnumRuleType.X, passThroughRule,
+							retriever.fReg.getDefaultPassThroughRuleFeatures(),
+							out);
 				}
-				threadPool.shutdown();
-				threadPool.awaitTermination(1, TimeUnit.DAYS);
-				// Add ascii constraints not already found in query
-				for (RuleWritable asciiConstraint : retriever.asciiConstraints) {
-					if (!retriever.foundAsciiConstraints
-							.contains(asciiConstraint)) {
-						features.writeRule(asciiConstraint,
-								AlignmentAndFeatureMap.EMPTY,
-								EnumRuleType.ASCII_OOV_DELETE, out);
-					}
+			}
+			// Add Deletion and OOV rules
+			Rule deletionRuleWritable = new Rule();
+			RuleString dr = new RuleString();
+			dr.add(Symbol.deserialise(-7));
```

Comment:
Is -7 and -6 an integer representation for DR and OOV? Shouldn't they be defined as constants somewhere?